core/modules/proxy/mysqlProxy.lua

```diff
@@ -529,26 +532,38 @@ function queryProcessing()
         local queryToPassProtect = "<![CDATA[" .. queryToPassStr .. "]]>"
         -- Wrap this in a pcall so that a meaningful error can
         -- be returned to the caller
-        -- xmlrpc.http.call(rpcHP, "submitQuery",
-        --                 queryToPassProtect, hintsToPassArr)
-        local callError, ok, res =
+        local pcall_status, xmlrpc_status, res =
            pcall(xmlrpc.http.call,
-                 rpcHP, "submitQuery", queryToPassProtect, hintsToPassArr)
-        if (not callError) then
-           return err.set(ERR_RPC_CALL, "Cannot connect to Qserv master; "
-                          .. "Exception in RPC call: " .. ok)
-        elseif (not ok) then
-           return err.set(ERR_RPC_CALL, "rpc call failed for " .. rpcHP)
+                 czarRpcUrl, "submitQuery", queryToPassProtect, hintsToPassArr)
+
+        -- if pcall failed then xmlrpc_status contains the related error message
+        -- if pcall succeed then xmlrpc_status contains the return code of
+        -- xmlrpc.http.call
+        if (not pcall_status) then
+            err_msg = xmlrpc_status
+            return err.set(ERR_RPC_CALL, "Unable to run lua xmlrpc client, message: " .. xmlrpc_status)
+        elseif (not xmlrpc_status) then
+            return err.set(ERR_RPC_CALL, "mysql-proxy RPC call failed for czar url: " .. czarRpcUrl)
         end
 
-        resultTableName = res[1]
-        msgTableName = res[2]
-        qservError = res[3]
-
-        if resultTableName == "error" then
+        qservError = res[1]
```

Comment:
This also contradicts documentation of `submitQuery()` method, I guess documentation is wrong.